app/main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name, health: int = 100, hidden: bool = False):
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        if self.health > 0:
+            self.alive.append(self)
+
+    def __bool__(self):
+        return self.health > 0
+
+    def __repr__(self):
+        return f"{{Name: {self.name}," \
+               f" Health: {self.health}," \
+               f" Hidden: {self.hidden}}}"
+
+    def dead(self):
+        if self in self.alive and self:
```

Comment:
if i understand your code correctly you remove animal from alive list if its health > 0 but should if health <= 0